specs/access-node-architecture/index.md

```diff
@@ -0,0 +1,79 @@
+
+# Detail level objectives
+
+- The handling of received raw protocol and execution state data is managed in the `Blockchain Data Service`. This state is provided to callers via the `Protocol API` and streamed to consumers of the `State Streamer`.
+- `Blockchain Data Service` state is accessed using `State Streamer Registration API` and via `State Streamer` clients.
+- `DPS` indexes execution state that has been delivered live from the protocol by `State Streamer`. The indexed state is used to answer script execution requests from the `DPS API`.
+
+# Design
+
+![architecture.jpg](images/architecture.jpg)
+
+# In detail
+
+To date the implementation of the Access Node has provided the following: 
+
+- Participates as a *staked* node within the Flow network
+- Participates in the private gossip network to follow protocol state
+- Enforces rate limits and throttling of requests
+- Handling of requests to the Access API from public callers for current live spork and
+    - Performs basic request validation
+    - Delegates script execution to upstream Execution Nodes
+    - Delegates transactions to Collection Nodes for processing by the network
+
+Script execution delegation to ENs is an obvious low hanging fruit limiting Access Node scalability and impacting transaction throughput capacity. In addition, scaling of Access API request handling is constrained by tight coupling of code within Access Node. The introduction of the DPS took the first step towards queryable locally held state that could be leveraged to scale Access Node, initially for protocol state only. Subsequently execution state sync provided the missing piece of the blockchain sync story. 
+
+## Revised ownership and responsibility
+
+By managing all chain synchronization concerns `Blockchain Data Service` provides a unified set of functionality that `Access Node` and `Observer Service` are built on, and which can be used as a standalone component for following trusted chain state more generally.
+
+The new `Access Node` builds on the original node capabilities above: 
+
+`Observer Service` is named as such due to it’s non-participation in the protocol
+
+`DPS` has two limited changes proposed: extraction of consensus follower into `Blockchain Data Service`. Then deprecating GCP based execution state sync and switching all chain sync logic over to `State Streamer`. 
+
+# Refactoring detailed diagrams
+### Starting Point
+![refactor-0.png](images/refactor-0.png)
+
+### Refactor Step 1
+![refactor-1.png](images/refactor-1.png)
+
+### Build `Blockchain Core Service`
+![refactor-2.png](images/refactor-2.png)
+
+### Build `Flow API Service`
+![refactor-3.png](images/refactor-3.png)
+
+# Additional questions
+
+- Historical access
+    - This re-architecture does not support historical state since consensus follower is a live only concept. But this remains one of the most often mentioned requests.
```

Comment:
We might want to reword this for the public - "While this re-architecture doesn't support access to historical state directly in this initial architecture revision, it makes support possible in future and we are working with community partners to achieve this"